### PR TITLE
OEL-1317: Improve navbar search form.

### PR DESCRIPTION
**Pull request by escuriola**

### CODE REVIEW

**Review by GilNovacomm — DISMISSED** [review]

File: modules/oe_whitelabel_search/src/Form/SearchForm.php
Comment (drishu) [inline modules/oe_whitelabel_search/src/Form/SearchForm.php]: lets add classes only in form--oe-whitelabel-search-form.html.twig

File: modules/oe_whitelabel_search/templates/form--oe-whitelabel-search-form.html.twig
Comment (drishu) [inline modules/oe_whitelabel_search/templates/form--oe-whitelabel-search-form.html.twig]: keep only d-flex mt-3 mt-lg-0

**Review by drishu — CHANGES_REQUESTED** [review]

File: templates/overrides/search/form-element--search-api-autocomplete.html.twig
Comment (drishu) [inline templates/overrides/search/form-element--search-api-autocomplete.html.twig]: if we make the changes I proposed then (from my tests) this is not needed anymore, but after you make the other changes please double check and let me know.

File: modules/oe_whitelabel_search/templates/block--oe-whitelabel-search-form.html.twig
Comment (drishu) [inline modules/oe_whitelabel_search/templates/block--oe-whitelabel-search-form.html.twig]: we should move all the templates from this folder to /templates/search

File: modules/oe_whitelabel_search/templates/form--oe-whitelabel-search-form.html.twig
Comment (drishu) [reply modules/oe_whitelabel_search/templates/form--oe-whitelabel-search-form.html.twig]: do we need "submittable" @escuriola ?

File: modules/oe_whitelabel_search/templates/block--oe-whitelabel-search-form.html.twig
Comment (drishu) [reply modules/oe_whitelabel_search/templates/block--oe-whitelabel-search-form.html.twig]: don't forget about this one

File: modules/oe_whitelabel_search/templates/block--oe-whitelabel-search-form.html.twig
Comment (drishu) [inline modules/oe_whitelabel_search/templates/block--oe-whitelabel-search-form.html.twig]: as we did in the oebt PR lets remove these as they are there in block.html.twig already

**Review by drishu — CHANGES_REQUESTED** [review]
2 issues I saw now
Screenshot 2022-04-14 at 10 04 03
we need a template for the navbar_right region and wrap {{ content}} in a div.nav
Screenshot 2022-04-14 at 10 04 15
we need to add class ps-0 to the site name because  we have to much passing on the left, or get rid of the "bcl-header__site-name" class from the link and only leave the one on the parent (closer to what is in netlify)

File: templates/search/block--oe-whitelabel-search-form.html.twig
Comment (donquixote) [inline templates/search/block--oe-whitelabel-search-form.html.twig:11]: The `{% block %}` is not really useful if we have no wrappers or other elements in the template.
In core, block templates without other stuff don't have `{% block %}`, e.g. `build/core/modules/system/templates/block--system-messages-block.html.twig`.
In whitelabel, we have `templates/overrides/navigation/block--languageswitcherinterfacetext.html.twig`, which does have the `{% block %}` syntax.

The only reason we would keep it is if we want to add wrapper elements or a title in the future, and we want inheriting templates to also get this change. But is this a realistic scenario?

I don't want to be the one who makes the final decision on this.

File: templates/overrides/navigation/block--system-branding-block.html.twig
Comment (donquixote) [inline templates/overrides/navigation/block--system-branding-block.html.twig:29]: The padding-left that we want to suppress with `ps-0` only applies in large viewports where the entire thing is hidden anyway. So do we need the `ps-0` at all?

Here is what I find in the site css (in showcase), coming from BCL:
```css
@media (min-width: 992px) {
    .bcl-header--ec .bcl-header__project.light,.bcl-header__project.light,.ec__header .bcl-header__project.light {
        background:#fff
    }
    .bcl-header__site-name {
        font-weight: 500;
        padding-left: 1.5rem
    }
```

**Comment by donquixote** [comment]
Seems mostly ok.
For the two points that I found I have no final opinion.
Either way we need to rebase.

**Review by drishu — DISMISSED** [review]

**Review by drishu — DISMISSED** [review]

File: modules/oe_whitelabel_search/src/Form/SearchForm.php
Comment (donquixote) [inline modules/oe_whitelabel_search/src/Form/SearchForm.php:56]: I think `flex-grow-1` would be more logical here. But `w-100` does the job, so let's keep it.

File: templates/overrides/navigation/block--system-branding-block.html.twig
Comment (donquixote) [reply templates/overrides/navigation/block--system-branding-block.html.twig:29]: I won't block the PR for this, but still would be interested in why we do this.

**Review by donquixote — COMMENTED** [review]
I am generally ok with the changes but I would wish some response to my questions.

File: modules/oe_whitelabel_search/templates/form--oe-whitelabel-search-form.html.twig
Comment (donquixote) [reply modules/oe_whitelabel_search/templates/form--oe-whitelabel-search-form.html.twig]: The "submittable" is gone. Resolving this.

File: templates/overrides/navigation/block--system-branding-block.html.twig
Comment (donquixote) [reply templates/overrides/navigation/block--system-branding-block.html.twig:29]: Actually the ps-0 is fine here, it is needed for EC header.

**Comment by donquixote** [comment]
Happy now, but need rebase.

**Review by drishu — DISMISSED** [review]

**Review by donquixote — DISMISSED** [review]

**Review by drishu — APPROVED** [review]

**Review by donquixote — APPROVED** [review]

**Pull request merged**
